app/src/main/java/org/android/go/sopt/ui/login/LoginActivity.kt

```diff
@@ -0,0 +1,112 @@
+import org.android.go.sopt.Week1Application
+import org.android.go.sopt.ui.signup.SignUpActivity
+import org.android.go.sopt.model.User
+import org.android.go.sopt.databinding.ActivityLoginBinding
+import org.android.go.sopt.util.*
+
+class LoginActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityLoginBinding
+    private lateinit var resultLauncher: ActivityResultLauncher<Intent>
+    private lateinit var userInfo: User
+    private var userRegisteredStatus: Boolean = false
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityLoginBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        // SharedPreferences에 저장된 유저 정보가 있으면 메인 화면으로 진입
+        if (isLastUserLoggedIn()) {
+            navigateToMainScreen()
+        }
+
+        binding.root.setOnClickListener {
+            hideKeyboard()
+        }
+
+        // 회원가입 화면으로부터 유저 정보 가져오기
+        resultLauncher = registerForActivityResult(
+            ActivityResultContracts.StartActivityForResult()
+        ) { result ->
+            if (result.resultCode == RESULT_OK) {
+                Snackbar.make(binding.root, SIGN_UP_SUCCESS_MSG, Snackbar.LENGTH_SHORT).show()
+                registerUserInfo(result.data)
+
+                // 자동 로그인을 위해 SharedPreferences에 유저 정보 저장하기
+                saveUserInfoToPrefs()
+            }
+        }
+
+        binding.btnLogin.setOnClickListener {
+            if (userRegisteredStatus) {
+                if (checkInputValues()) {
+                    Toast.makeText(this, LOGIN_SUCCESS_MSG, Toast.LENGTH_SHORT).show()
+                    navigateToMainScreen()
+                } else {
+                    Toast.makeText(this, LOGIN_FAIL_MSG, Toast.LENGTH_SHORT).show()
+                }
+            } else {
+                Toast.makeText(this, NOT_YET_REGISTERED_MSG, Toast.LENGTH_SHORT).show()
+            }
+        }
+
+        binding.btnSignUp.setOnClickListener {
+            val intent = Intent(this, SignUpActivity::class.java)
+            resultLauncher.launch(intent)
+        }
+    }
+
+    private fun isLastUserLoggedIn(): Boolean {
+        val id = Week1Application.prefs.getString(ID_KEY, null)
+        val pw = Week1Application.prefs.getString(PW_KEY, null)
+        val name = Week1Application.prefs.getString(NAME_KEY, null)
+        val hobby = Week1Application.prefs.getString(HOBBY_KEY, null)
+        return id != null && pw != null && name != null && hobby != null
+    }
+
+    private fun hideKeyboard() {
+        val imm = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+        imm.hideSoftInputFromWindow(binding.root.windowToken, 0)
+    }
+
+    private fun navigateToMainScreen() {
+        val intent = Intent(this, MainActivity::class.java)
+        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK or Intent.FLAG_ACTIVITY_NEW_TASK)
+        startActivity(intent)
```

Comment:
```suggestion
        Intent(this, MainActivity::class.java).apply {
                    addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK or Intent.FLAG_ACTIVITY_NEW_TASK)
                    startActivity(this)
                }
```
스코프 함수 `apply`나 `also`를 활용하여 코틀린스럽게 작성할 수도 있습니다 ㅎㅎ